### PR TITLE
cert: Add test exclusion in cert tasks

### PR DESCRIPTION
**Pull request by kosstennbl**
## Description
Adds exclude="... ..." parameter for cert tasks in place of the previous ~... exclusion pattern.
Minor refactor of cert tasks.

Check issue for more info.

## Issues:
Refs: #2048

## Additional:
Build and functionality tested, documentation added.

### CODE REVIEW

File: src/tasks/cert/cert.cr [hunk 1]
```diff
@@ -45,3 +45,26 @@ task "cert", ["version", "cert_compatibility", "cert_state", "cert_security", "c
   stdout_info "Results have been saved to #{CNFManager::Points::Results.file}".colorize(:green)
 end
 
+def get_excluded_tasks(args)
+  if args.named.has_key? "exclude"
+    exclude = args.named["exclude"]
+    if exclude.is_a? String
```
Comment (martin-mat) [inline src/tasks/cert/cert.cr:51]: can also a check for correct tasks/tests be checked? (i.e. exclusion set contains cert tests - based on points.yml - and nothing else).
Also, not sure whether this file is a good place for such 2 functions, requiring this task-oriented file to be included from elsewhere. Consider relocating it elsewhere.

**Review by martin-mat — CHANGES_REQUESTED** [review]
Please check just one comment, other than that OK.

File: src/tasks/cert/cert.cr (hunk 1 above)
Comment (kosstennbl) [reply src/tasks/cert/cert.cr]: Agree, done.

File: src/tasks/cert/cert_compatibility.cr [hunk 2]
```diff
@@ -4,21 +4,23 @@ require "file_utils"
...
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
-
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
+  
   stdout_score(tags, "Compatibility, Installability, and Upgradeability")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_compatibility.cr:21]: do not copy/paste.
Perhaps the best would be to add a param and include that into stdout_score.

File: src/tasks/cert/cert_configuration.cr [hunk 3]
```diff
@@ -5,23 +5,24 @@ require "colorize"
...
   tags << "essential" if essential_only
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   stdout_score(tags, "configuration")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_configuration.cr:23]: copy/paste

File: src/tasks/cert/cert_microservice.cr [hunk 4]
```diff
@@ -7,21 +7,23 @@ require "../utils/utils.cr"
...
   tags << "essential" if essential_only
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   stdout_score(tags, "microservice")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_microservice.cr:24]: copy/paste

File: src/tasks/cert/cert_observability.cr [hunk 5]
```diff
@@ -4,21 +4,23 @@ require "file_utils"
...
   tags << "essential" if essential_only
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   stdout_score(tags, "Observability and Diagnostics")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_observability.cr:21]: copy/paste

File: src/tasks/cert/cert_resilience.cr [hunk 6]
```diff
@@ -3,24 +3,26 @@ require "sam"
...
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   Log.for("verbose").info {  "resilience" } if check_verbose(args)
   VERBOSE_LOGGING.debug "resilience args.raw: #{args.raw}" if check_verbose(args)
   VERBOSE_LOGGING.debug "resilience args.named: #{args.named}" if check_verbose(args)
   stdout_score(tags, "Reliability, Resilience, and Availability")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_resilience.cr:23]: copy/paste

File: src/tasks/cert/cert_security.cr [hunk 7]
```diff
@@ -4,21 +4,23 @@ require "file_utils"
...
   tags << "essential" if essential_only
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   stdout_score(tags, "security")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_security.cr:21]: copy/paste

File: src/tasks/cert/cert_state.cr [hunk 8]
```diff
@@ -5,21 +5,23 @@ require "colorize"
...
   tags << "essential" if essential_only
 
-  tasks = CNFManager::Points.tasks_by_tag_intersection(tags)
-  tasks.each do |task|
-    t.invoke(task)
-  end
+  invoke_tasks_by_tag_list(t, tags, exclude_tasks: exclude)
 
   stdout_score(tags, "state")
+  if !exclude.empty?
```
Comment (martin-mat) [inline src/tasks/cert/cert_state.cr:22]: copy/paste

**Review by martin-mat — CHANGES_REQUESTED** [review]

File: src/tasks/cert/cert_compatibility.cr (hunk 2 above)
Comment (kosstennbl) [reply src/tasks/cert/cert_compatibility.cr]: Fixed, added new function that will call stdout_score and then add the warning if needed.

**Comment by martin-mat** [comment]
github actions passed

https://github.com/cnti-testcatalog/testsuite/actions/runs/9409085660

**Comment by horecoli** [comment]
LGTM 👍

**Pull request merged**